lib/appraisal/matrix/extensions/appraisal_file.rb

```diff
@@ -5,11 +5,28 @@
 
 module Appraisal::Matrix
   module AppraiseFileWithMatrix
-    include RubygemsHelper
+
+    class VersionArray
+      SUPPORTED_VERSION_STEPS = [:major, :minor, :patch].freeze
+
+      attr_reader :gem_name, :version_restrictions, :step
+
+      def initialize(gem_name:, versions:, step: :minor)
+        SUPPORTED_VERSION_STEPS.include?(step) or raise("Unsupported version step: #{step}")
+
+        @gem_name = gem_name
+        @version_restrictions = versions.map { |version| Gem::Dependency.new('', version) }
```

Comment:
Looks like there is a `Gem::Requirement` class that we can use with just the version!